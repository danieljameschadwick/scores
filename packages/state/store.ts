import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import AsyncStorage from '@react-native-async-storage/async-storage';
import { persistReducer, persistStore } from "redux-persist";
import thunk from "redux-thunk";
import themeReducer from "./reducer/ThemeReducer";

const persistConfig = {
  key: "root",
  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, themeReducer);

const store = configureStore({
  reducer: persistedReducer,
  devTools: process.env.NODE_ENV !== "production",
  middleware: [thunk],
});

export const persistor = persistStore(store);

export type AppState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export type AppThunk<ReturnType = void> = ThunkAction<ReturnType,
  AppState,
  unknown,
  Action<string>>;

export default store;
