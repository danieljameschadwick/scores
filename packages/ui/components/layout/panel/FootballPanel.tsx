import React, { useEffect, useState } from "react";
import { StyleSheet, Text, View } from "react-native";
import IonIcon from "react-native-vector-icons/Ionicons";
import { Panel } from "@scores/ui/components/layout/panel/Panel";
import { getPrimaryText } from "@scores/theme/utils/variables";
import { GAME_TYPE } from "@scores/types/enum/GameType";
import { PanelFixture } from "@scores/ui/components/layout/panel/fixture/PanelFixture";
import { getFixtures } from "@scores/http/services/football";
import { normaliseScores } from "@scores/http/utils/normaliseScores";
import { useTheme } from "@scores/theme/utils/theme";

export const FootballPanel = () => {
  const themeStyles = useTheme();
  const [fixtures, setFixtures] = useState<{}>(null);

  useEffect(() => {
    const fetchData = async () => {
      setFixtures(normaliseScores(await getFixtures(), GAME_TYPE.FOOTBALL));
    };

    fetchData();
  }, []);

  if (!fixtures) {
    return (
      <Panel
        title={"Football"}
        icon={<IonIcon name={"football"} size={24} color={getPrimaryText()} />}
        testID={"football-panel"}
      >
        <View style={[styles.container]}>
          <Text style={[themeStyles.text]} testID={"loading"}>
            Loading
          </Text>
        </View>
      </Panel>
    );
  }

  return (
    <Panel
      title={"Football"}
      icon={<IonIcon name={"football"} size={24} color={getPrimaryText()} />}
      testID={"football-panel"}
    >
      {fixtures && (
        <View style={[styles.container]} testID={"fixtures"}>
          {Object.keys(fixtures)
            .sort()
            .map((key) => {
              return fixtures[key].map((score) => {
                const { id } = score;

                return <PanelFixture key={id} {...score} />;
              });
            })}
        </View>
      )}
    </Panel>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
  },
});
