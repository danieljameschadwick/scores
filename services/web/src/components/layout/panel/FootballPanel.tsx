import React from "react";
import { StyleSheet, View } from "react-native-web";
import IonIcon from "react-native-vector-icons/Ionicons";
import { Panel } from "@src/components/layout/panel/Panel";
import { getPrimaryText } from "@scores/theme/utils/variables";
import { GAME_TYPE } from "@src/enum/GameType";
import { PanelFixture } from "@src/components/layout/panel/fixture/PanelFixture";

import footballFixtures from "@src/mocks/footballFixtures";
import { normaliseScores } from "@src/utils/scores/normaliseScores";

export const FootballPanel = () => {
  const fixtures = normaliseScores(footballFixtures, GAME_TYPE.FOOTBALL);

  return (
    <Panel
      title={"Football"}
      icon={<IonIcon name={"football"} size={24} color={getPrimaryText()} />}
    >
      <View style={[styles.container]}>
        {Object.keys(fixtures)
          .sort()
          .map((key) => {
            return (
              <>
                {fixtures[key].map((score) => {
                  const { id } = score;

                  return <PanelFixture key={id} {...score} />;
                })}
              </>
            );
          })}
      </View>
    </Panel>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
  },
});
