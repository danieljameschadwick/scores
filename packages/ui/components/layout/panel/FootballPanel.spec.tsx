import React from "react";
import { render, screen, waitFor } from "@testing-library/react-native";
import { Provider } from "react-redux";
import store, { persistor } from "@scores/state/store";
import { PersistGate } from "redux-persist/integration/react";
import { FootballPanel } from "./FootballPanel";

const customRender = () => {
  return render(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <FootballPanel />
      </PersistGate>
    </Provider>
  );
};

describe("FootballPanel component", () => {
  it("renders base loading component", async () => {
    const { queryByTestId } = customRender();

    expect(queryByTestId("loading")).toBeTruthy();
    
    waitFor(() => expect(queryByTestId("panel-fixture")).toBeTruthy());

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("renders loaded component", async () => {
    const { queryByTestId } = customRender();

    expect(queryByTestId("loading")).toBeTruthy();
    
    waitFor(() => expect(queryByTestId("panel-fixture")).toBeTruthy());

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
