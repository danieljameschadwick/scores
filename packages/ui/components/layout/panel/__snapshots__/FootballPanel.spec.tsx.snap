// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FootballPanel component renders base loading component 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderRadius": 10,
          "marginBottom": 10,
          "paddingBottom": 16,
          "paddingHorizontal": 16,
          "paddingTop": 8,
        },
      ],
      Object {
        "backgroundColor": "rgb(232, 234, 237)",
        "borderColor": "rgb(215, 220, 224)",
      },
    ]
  }
  testID="football-panel"
>
  <View
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#dcdddf",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 10,
            "paddingBottom": 5,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 3,
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "800",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      Football
    </Text>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
          ]
        }
        testID="loading"
      >
        Loading
      </Text>
    </View>
  </View>
</View>
`;

exports[`FootballPanel component renders loaded component 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderRadius": 10,
          "marginBottom": 10,
          "paddingBottom": 16,
          "paddingHorizontal": 16,
          "paddingTop": 8,
        },
      ],
      Object {
        "backgroundColor": "rgb(232, 234, 237)",
        "borderColor": "rgb(215, 220, 224)",
      },
    ]
  }
  testID="football-panel"
>
  <View
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "borderBottomColor": "#dcdddf",
            "borderBottomWidth": 1,
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 10,
            "paddingBottom": 5,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginRight": 3,
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 13,
            "fontWeight": "800",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#000",
          },
        ]
      }
    >
      Football
    </Text>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
          ]
        }
        testID="loading"
      >
        Loading
      </Text>
    </View>
  </View>
</View>
`;
